jaeger_client/codecs.py

```diff
@@ -271,21 +271,25 @@ def inject(self, span_context, carrier):
         carrier[self.span_header] = format(span_context.span_id, 'x').zfill(16)
         if span_context.parent_id is not None:
             carrier[self.parent_span_header] = format(span_context.parent_id, 'x').zfill(16)
-        carrier[self.flags_header] = str(span_context.flags)
+        if span_context.flags & DEBUG_FLAG == DEBUG_FLAG:
+            carrier[self.flags_header] = '1'
+        elif span_context.flags & SAMPLED_FLAG == SAMPLED_FLAG:
+            carrier[self.sampled_header] = '1'
 
     def extract(self, carrier):
         if not isinstance(carrier, dict):
             raise InvalidCarrierException('carrier not a dictionary')
-        trace_id = header_to_hex(carrier.get(self.trace_header.lower()))
-        span_id = header_to_hex(carrier.get(self.span_header.lower()))
-        parent_id = carrier.get(self.parent_span_header.lower(), None)
+        ic_map = dict([(k.lower(), k) for k in carrier])
```

Comment:
what does `ic` stand for?

Reply:
"Ignore case".  Updated to `lowercase_keys` for clarity.